### DISCUSSION TITLE
Logs panel: specify field selection

### DISCUSSION THREAD

**Discussion by mieliespoor** [grafana/grafana#38290] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
On the logs panel it should be possible to select which fields from the log item is being shown. Sometimes, the full log message is not needed as it doesn't make sense to show all that information on a dashboard.

**Why is this needed**:
We have a query showing Cloudwatch Log data and a query similar to:

```
fields @timestamp, Level, Properties.CorrelationId, MessageTemplate
| sort @timestamp desc
| limit 20
```

When running the same query in Grafana, it will only show the data from the Level field and nothing else. The expectation is that we see the data for all the fields listed in the query.

**Comment by bishwashrestha** [comment]
Yes please! :)

**Comment by aocenas** [comment]
I think this use case would make more sense with table panel.

Also one thing you can use is Toggle detected fields when in explore, see: https://grafana.com/docs/grafana/latest/explore/logs-integration/. 

Do you want to use this in dashboards? Is there a reason you want to use logs panel instead of table panel?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by jan-xyz** [comment]
> Do you want to use this in dashboards? Is there a reason you want to use logs panel instead of table panel?

The logs panel has a much better support for things like data links, to follow directly from a trace ID into jaeger for example.

**Comment by mbrevda** [comment]
+1 for this. The flexibility in editing a line (say, via promtail) is great, but the result is permanently altering the line and all data in it. For example, when extracting fields from a json log, all non-extracted fields are lost. It would be great if there was some way to control what was visible by default vs which labels/fields loki would pick up on.

**Comment by dylan-smith-wolf** [comment]
Something like this would be great.  Currently we have a query like:
```
fields @timestamp, @message
| sort @timestamp desc
| limit 20
```
The @message is the full log message in JSON, and we use that so that we can see the details.  It would be awesome if we could choose which fields within the JSON to show in the logs list, and then all fields would be detected and shown in the details.  Currently "Enable log details" shows the same thing as the log line.  We condense the log line by not prettifying and not wrapping, but it would make more sense to be able to just have one or two key fields shown for the line, and the full details when expanded.

**Comment by burmajam** [comment]
When importing logs from InfluxDB2, we have multiple fields. In order to show them as tags we have to pivot streams. After that point, there's no way to say which column is log itself (like we can do for `level` with renaming a column). Each refresh shows different column as a log.

**Comment by enkelmedia** [comment]
I'm kind of a newbie but I think that I'm looking for the same thing. Basically being able to replace the "whole" log messages (in my case structured json) and only show the "Message"-property from the JSON in the panel.

![image](https://github.com/grafana/grafana/assets/1782524/80a327b0-e976-469a-9bc1-b66326dec992)

Someone mentioned a table to be the right solution but this would love the "Enable log details"-toggle which is really nice.

**UPDATE:**

Turns out that I could do this by using [line_format](https://grafana.com/docs/loki/latest/logql/log_queries/#log-pipeline). This would basically extract the "Message"-property from the JSON and use in the list.

```
{app="myapp"} |= `` | json | line_format "{{.Message}}"
```

![image](https://github.com/grafana/grafana/assets/1782524/f7c3be17-a7ee-42eb-be75-c213c812e636)

**Reply by mbrevda** [reply]
when you click the `>` to expand, do you still get the complete payload?

**Reply by enkelmedia** [reply]
Yes :)

**Reply by CumpsD** [reply]
Sadly enough this only works for loki, doesnt work for cloudwatch :(

**Comment by ivanahuckova** [comment]
For additional discussion, please use https://github.com/grafana/grafana/issues/77552
